src/parser/index.ts

```diff
@@ -145,31 +145,46 @@ export class Parser {
 
     /**
      * Simplify the logical expression if possible.
+     * - AND, OR with one operand is collapsed.
+     * - Logical expressions without operands are treated as collection.
+     * @param collapsible
```

Comment:
Also a little lost by what `collapsible` is conveying.

Reply:
219b9f5a280fff7c8719b3b952ed3a352278f47a